libs/scripts/personalization.js

```diff
@@ -0,0 +1,374 @@
+  const s = str.toLowerCase();
+  return s.replace(RE_KEY_REPLACE, '');
+}
+
+function transformKeys(obj) {
+  return Object.keys(obj).reduce((newObj, key) => {
+    newObj[toLowerAlpha(key)] = obj[key];
+    return newObj;
+  }, {});
+}
+
+export function parseConfig(data) {
+  if (!data?.length) return null;
+
+  const config = {};
+  const experiences = data.map((d) => transformKeys(d));
+
+  try {
+    const variants = {};
+    const variantNames = Object.keys(experiences[0])
+      .filter((vn) => !MANIFEST_KEYS.includes(vn));
+
+    variantNames.forEach((variantName) => {
+      variants[variantName] = { commands: [] };
+    });
+
+    const currentPath = new URL(window.location).pathname;
+
+    experiences.forEach((line) => {
+      const action = line.action?.toLowerCase();
+      const { selector } = line;
+      const pageFilter = line['page filter'] || line['page filter optional'];
+
+      if (pageFilter && !matchGlob(pageFilter, currentPath)) return;
+
+      variantNames.forEach((vn) => {
+        if (!line[vn]) return;
+
+        const variantInfo = {
+          action,
+          selector,
+          pageFilter,
+          target: line[vn],
+        };
+
+        if (GLOBAL_CMDS.includes(action)) {
+          variants[vn][action] = variants[vn][action] || [];
+          variants[vn][action].push({
+            selector: normalizePath(selector),
+            val: normalizePath(line[vn]),
+          });
+        } else if (VALID_COMMANDS.includes(action)) {
+          variants[vn].commands.push(variantInfo);
+        } else {
+          console.log('Invalid action found: ', line);
+        }
+      });
+    });
+
+    config.variants = variants;
+    config.variantNames = variantNames;
+    return config;
+  } catch (e) {
+    console.log('error parsing personalization config:', e, experiences);
+  }
+  return null;
+}
+
+export async function replaceInner(path, element) {
+  if (!path || !element) return false;
+  let plainPath = path.endsWith('/') ? `${path}index` : path;
+  plainPath = plainPath.endsWith('.plain.html') ? plainPath : `${plainPath}.plain.html`;
+  try {
+    const resp = await fetch(plainPath);
+    if (!resp.ok) {
+      throw new Error('Invalid response', resp);
+    }
+    const html = await resp.text();
+    element.innerHTML = html;
+    return true;
+  } catch (e) {
+    console.log(`error loading experiment content: ${plainPath}`, e);
+  }
+  return false;
+}
+
+function getPersonalizationVariant(variantNames = [], variantLabel = null) {
+  const tagNames = Object.keys(PERSONALIZATION_TAGS);
+  // handle multiple variants that are space / comma delimited
+  const matchingVariant = variantNames.find((variant) => {
+    const names = variant.split(RE_SPACE_COMMA).filter(Boolean);
+    if (names.some((name) => (
+      name === variantLabel || (tagNames.includes(name) && PERSONALIZATION_TAGS[name]())))
+    ) {
+      return variant;
+    }
+    return false;
+  });
+  return matchingVariant;
+}
```

Comment:
This can be simplified (untested code below)
```js
function getPersonalizationVariant(variantNames = [], variantLabel = null) {
  const tagNames = Object.keys(PERSONALIZATION_TAGS);
  const matchingVariant = variantNames.find((variant) => {
    const names = variant.split(RE_SPACE_COMMA).filter(Boolean);
    // Simplified conditional logic by using implicit return
    return names.some((name) => name === variantLabel || (tagNames.includes(name) && PERSONALIZATION_TAGS[name]()));
  });
  return matchingVariant;
}
```